File: angular/src/app/base-elements/form/form-field/models/FormField.ts
import { FieldType } from './FieldType';
export class FormField {
  nameField: string;
  options?: Option[];
  type?: FieldType;
  validations?: IValidation;
  showProperties?: IShowProperty;
}

export class IShowProperty {
  readOnly: boolean;
  hidden: boolean;
  label: string;
}

export interface IValidation {
  required: boolean;
  maxLength: number;
  minLength: number;
  max: number;
  min: number;
  regExp: string;
}

export class Option {
  label: string;
  value: number;

  constructor(value: number, label: string) {
    this.label = label;
    this.value = value;
  }
}

File: ServiceDesk.SmartUI.Forms/FormField.cs
using ServiceDesk.SmartUI.Forms.FormFieldsBuilder.OptionBuilders;
using System;
using System.Collections.Generic;
using System.Text;

namespace ServiceDesk.SmartUI.Forms
{
    public class FormField
    {
        public string NameField { get; set; }
        public IEnumerable<Option> Options { get; set; }
        public string Type { get; set; }
        public Validation Validations { get; set; }
        public ShowProperty ShowProperties { get; set; }
    }

    public class ShowProperty
    {
        public bool ReadOnly { get; set; }
        public bool Hidden { get; set; }
        public string Label { get; set; }
    }

    public class Validation
    {
        public bool? Required { get; set; }
        public int? MaxLength { get; set; }
        public int? MinLength { get; set; }
        public int? Max { get; set; }
        public int? Min { get; set; }
        public string RegExp { get; set; }

    }

}
